Checkout/Block/Form/Payments.php

```diff
@@ -141,12 +148,52 @@ public function getIframeUrl()
     private function getStyles()
     {
         $styles = Mage::getModuleDir('data', 'Bold_Checkout') . DS . 'form/payments/styles.json';
-        // @phpcs:disable MEQP1.Security.DiscouragedFunction.Found
         if (file_exists($styles)) {
             return json_decode(file_get_contents($styles), true);
         }
-        // @phpcs:enable MEQP1.Security.DiscouragedFunction.Found
-
         return null;
     }
+
+    /**
+     * Refresh order on Bold side before render PIGI iframe.
+     *
+     * @return void
+     * @throws Mage_Core_Exception
+     */
+    private function refreshOrder()
+    {
+        $session = Mage::getSingleton('checkout/session');
+        $boldCheckoutData = $session->getBoldCheckoutData();
+        if (!$boldCheckoutData) {
+            return;
+        }
+        $boldEmailAddress = $boldCheckoutData->data->application_state->customer->email_address;
+        $email = $session->getQuote()->getBillingAddress()->getEmail();
+        $firstname = $session->getQuote()->getBillingAddress()->getFirstname();
+        $lastname = $session->getQuote()->getBillingAddress()->getLastname();
+        if ($this->customerIsGuest() && !$boldEmailAddress && $email && $firstname && $lastname) {
+            $guestResult = Bold_Checkout_StorefrontClient::call(
+                'POST',
+                'customer/guest',
```

Comment:
We may need to look into a different solution for this. Making requests to storefront api's from the backend is going to run into rate limiting issues. https://developer.boldcommerce.com/guides/getting-started/rate-limiting

Since storefront api's are limiting based on user (IP address) once they're made from the server it will all count as one user.

If there's no other way to do this for now we may need to investigate this outside of this change to see if we can get around the rate limiting issue.

Reply:
Removed.